### PR TITLE
Make Sphinx calls more resilient to errors

### PR DESCRIPTION
**Pull request by ovan**
* Change listingindex to expect search adapters to return Result types
* Adapt existing sphinx adapter to return Result types
* Add timeout to sphinx queries
* Handle sphinx errors, such as no connection to server, by wrapping to Result::Error
* Log errors before returning from service and show err msg in UI

### CODE REVIEW

File: app/services/listing_index_service/api/api.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@ class << self
     end
 
     def self.listings
-      self.listings_api ||= ListingIndexService::API::Listings.new
+      self.listings_api ||= ListingIndexService::API::Listings.new(Rails.logger)
```
Comment (rap1ds) [inline app/services/listing_index_service/api/api.rb]: You could (and should?) use the new and shiny SharetribeLogger.

```ruby
ListingIndexService::API::Listings.new(SharetribeLogger.new(:listing_index_service))
```

File: app/services/listing_index_service/search/sphinx_adapter.rb [hunk 2]
```diff
@@ -105,10 +113,15 @@ def search_with_sphinx(community_id:, search:, included_models:)
...
-          order: 'sort_date DESC'
+          order: 'sort_date DESC',
+          max_query_time: 500 # Timeout and fail after 500ms
         )
 
-        {count: models.total_entries, listings: models}
+        begin
+          success_result(models.total_entries, models, includes)
+        rescue ThinkingSphinx::SphinxError => e
+          Result::Error.new(e)
```
Comment (rap1ds) [inline app/services/listing_index_service/search/sphinx_adapter.rb]: This is very nice! 

I have seen two types of error thrown by Sphinx that we should catch:

1. In development, Sphinx is not running.
2. In production, Sphinx is reindexing.

Have you tested does this now catch them both? I.e. have you tested that the exception thrown from those two cases are `ThinkingSphinx::SphinxError` or subclasses of it?

**Comment by rap1ds** [comment]
Maybe not part of this pull request, but the flash notification for Resut::Error is now "Something went wrong", in a red background.

If the error occured becase the developer didn't have Sphinx running, we could be more informative and say "Could not connect to the search engine", for example. If Sphinx is reindexing in the production, we could say "Search engine is currently unavailable. Please try again in a minute"

File: app/services/listing_index_service/api/api.rb (hunk 1 above)
Comment (ovan) [inline app/services/listing_index_service/api/api.rb]: kewl, I was thinking about that but didn't realize it's already in master. Will definitely change.

File: app/services/listing_index_service/search/sphinx_adapter.rb (hunk 2 above)
Comment (ovan) [inline app/services/listing_index_service/search/sphinx_adapter.rb]: I haven't tested the indexing thing but this is what the Sphinx-documentation says, all errors raised are SphinxErrors. The case of sphinx not running in development is just a connection error, so it's indistinguishable from a network error in production or production sphinx being down.

**Pull request merged**